actionview/test/template/form_helper_test.rb

```diff
@@ -2461,6 +2461,34 @@ def test_nested_fields_for
     assert_dom_equal expected, output_buffer
   end
 
+  def test_nested_fields_for_field_id
+    form_for(@post) do |form|
+      concat form.fields_for(:comment_attributes, @comment) { |c|
+        concat c.field_id(:body)
+      }
+    end
+
+    expected = whole_form("/posts/123", "edit_post_123", "edit_post", method: "patch") do
+      "post_comment_attributes_body"
+    end
+
+    assert_dom_equal expected, output_buffer
+  end
+
+  def test_nested_fields_for_field_id_with_index_override
+    form_for(@post) do |form|
+      concat form.fields_for(:comment_attributes, @comment, index: 1) { |c|
+        concat c.field_id(:body)
+      }
+    end
+
+    expected = whole_form("/posts/123", "edit_post_123", "edit_post", method: "patch") do
+      "post_comment_attributes_1_body"
```

Comment:
Just curious....
What would be the expected value before this change?

Reply:
My expectations around calling `field_id` on a nested form would be that the fields' `[id]` attributes would be namespaced to account for both outer form names and inner form `index` values.

Prior to this change, the generated value is `comment_1_body`, which omits both the outer model's `model_name` (i.e. `post`), and the scopes the attributes by the nested model's `model_name` (i.e. `comment`) instead of the nested attributes (i.e. `comment_attributes`).